tawazi/node/node.py

```diff
@@ -423,8 +424,14 @@ def __init__(
             setup (bool): Look at ExecNode's Documentation
             unpack_to (Optional[int]): Look at ExecNode's Documentation
         """
+        if isinstance(func, partial):
+            func = functools.update_wrapper(func, func.func)
+
+        if not hasattr(func, "__qualname__"):
+            func.__qualname__ = "__anonymous__"
```

Comment:
is this still necessary ?

Reply:
hmm, it might but tbh I'd rather encounter a bug because it may induce weird unexpected behaviour with edge cases objects that could actually be worthy of an ege case in the node constructor by themselves

Reply:
I also removed the additional point in the doc since we actually don't want to have a different documented behaviour for partial fns